ocs_ci/utility/utils.py

```diff
@@ -1302,25 +1302,104 @@ def get_az_count():
         return 1
 
 
-def ceph_health_check(namespace=None, tries=20, delay=30):
+def ceph_health_check(
+    namespace=None, tries=20, delay=30, intelligent_wait=False,
+    max_timeout_for_pg=1500,
+):
     """
     Args:
         namespace (str): Namespace of OCS
             (default: config.ENV_DATA['cluster_namespace'])
         tries (int): Number of retries
         delay (int): Delay in seconds between retries
+        intelligent_wait (bool): If true, the ceph_health is waiting for
+            HEALTH_OK if we have degraded or undersized PGs. In case the time
+            between decrease of degraded/undersized PGs is less than
+            max_timeout_for_pg than we continue to wait.
+        max_timeout_for_pg (int): Max timout in seconds to wait for decrease in
```

Comment:
Somehow the name ```max_timeout_for_pg``` is bit misleading to me, it made me think like 'its timeout per pg' . How about ```pg_recovery_progress_timeout```. May be others would be having better name , wdyt?